drivers/ieee802154/ieee802154_cc1101.c

```diff
@@ -0,0 +1,873 @@
+		SYS_LOG_ERR("Cannot switch to TX mode");
+		goto out;
+	}
+
+	if (!write_txfifo(cc1101, &len, CC1101_PHY_HDR_LEN) ||
+	    !write_txfifo(cc1101, frame, len)) {
+		SYS_LOG_ERR("Cannot fill-in TX fifo");
+		goto out;
+	}
+
+	atomic_set(&cc1101->tx, 1);
+	atomic_set(&cc1101->tx_start, 0);
+
+	if (!instruct_stx(cc1101)) {
+		SYS_LOG_ERR("Cannot start transmission");
+		goto out;
+	}
+
+	/* Wait for SYNC to be sent */
+	k_sem_take(&cc1101->tx_sync, 100);
+	if (atomic_get(&cc1101->tx_start) == 1) {
+		/* Now wait for the packet to be fully sent */
+		k_sem_take(&cc1101->tx_sync, 100);
+	}
+
+out:
+	_cc1101_print_status(get_status(cc1101));
+
+	if (atomic_get(&cc1101->tx) == 1 &&
+	    get_rx_bytes(cc1101) != 0) {
+		SYS_LOG_ERR("TX Failed");
+
+		atomic_set(&cc1101->tx_start, 0);
+		instruct_sftx(cc1101);
+		status = false;
+	} else {
+		status = true;
+	}
+
+	atomic_set(&cc1101->tx, 0);
+
+	/* Get back to RX */
+	instruct_srx(cc1101);
+
+	return status ? 0 : -EIO;
+}
+
+static int cc1101_start(struct device *dev)
+{
+	struct cc1101_context *cc1101 = dev->driver_data;
+
+	if (!instruct_sidle(cc1101) ||
+	    !instruct_sftx(cc1101) ||
+	    !instruct_sfrx(cc1101) ||
+	    rf_calibrate(cc1101)) {
+		SYS_LOG_ERR("Could not proceed");
+		return -EIO;
+	}
+
+	enable_gpio_interrupt(cc1101, CC1101_GPIO_IDX_GPIO0, true);
+	enable_gpio_interrupt(cc1101, CC1101_GPIO_IDX_GPIO1, true);
+
+	_cc1101_print_status(get_status(cc1101));
+
+	return 0;
+}
+
+static int cc1101_stop(struct device *dev)
+{
+	struct cc1101_context *cc1101 = dev->driver_data;
+
+	enable_gpio_interrupt(cc1101, CC1101_GPIO_IDX_GPIO0, false);
+	enable_gpio_interrupt(cc1101, CC1101_GPIO_IDX_GPIO1, false);
+
+
+	if (!instruct_spwd(cc1101)) {
+		SYS_LOG_ERR("Could not proceed");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static u16_t cc1101_get_channel_count(struct device *dev)
+{
+	struct cc1101_context *cc1101 = dev->driver_data;
+
+	return cc1101->rf_settings->channel_limit;
+}
+
+/******************
+ * Initialization *
+ *****************/
+
+static int power_on_and_setup(struct device *dev)
+{
+	struct cc1101_context *cc1101 = dev->driver_data;
+
+	if (!instruct_sres(cc1101)) {
+		SYS_LOG_ERR("Cannot reset");
+		return -EIO;
+	}
+
+	if (!rf_install_settings(dev, &cc1101_rf_settings)) {
+		SYS_LOG_ERR("Cannot write settings");
+		return -EIO;
+	}
+
+	if (!write_reg_iocfg2(cc1101, CC1101_SETTING_IOCFG2) ||
+	    !write_reg_iocfg1(cc1101, CC1101_SETTING_IOCFG1) ||
+	    !write_reg_iocfg0(cc1101, CC1101_SETTING_IOCFG0)) {
+		SYS_LOG_ERR("Cannot configure GPIOs");
+		return -EIO;
+	}
+
+	setup_gpio_callback(dev);
+
+	return rf_calibrate(cc1101);
+}
+
+static int configure_spi(struct device *dev)
+{
+	struct cc1101_context *cc1101 = dev->driver_data;
+
+	cc1101->spi = device_get_binding(
+			CONFIG_IEEE802154_CC1101_SPI_DRV_NAME);
+	if (!cc1101->spi) {
+		SYS_LOG_ERR("Unable to get SPI device");
+		return -ENODEV;
+	}
+
+	if (IS_ENABLED(CONFIG_IEEE802154_CC1101_GPIO_SPI_CS)) {
```

Comment:
Have you tried to build with that CONFIG option being unset? I don't think it will built because cs_ctrl will be undefined. (bug is also present in cc1200 obviously)